expectations.go

```diff
@@ -130,28 +130,58 @@ func (e *commonExpectation) String() string {
 
 // queryBasedExpectation is a base class that adds a query matching logic
 type queryBasedExpectation struct {
-	expectSQL string
-	args      []interface{}
-}
+	expectSQL          string
+	expectRewrittenSQL string
+	args               []interface{}
+}
+
+func (e *queryBasedExpectation) argsMatches(sql string, args []interface{}) (rewrittenSQL string, err error) {
+	eargs := e.args
+	// check for any query rewriters
+	// according to current pgx docs, a QueryRewriter is only supported as the first
+	// argument.
+	if len(args) == 1 {
+		if qrw, ok := args[0].(pgx.QueryRewriter); ok {
+			// note: pgx.Conn is not currently used by the query rewriter, but is part
+			// of the method signature, so just create an empty pointer for now.
+			rewrittenSQL, newArgs, err := qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```

Comment:
suggestion (if-minor): Possibly reduce chance for future bugs

#### Detail
Which variables in the [short variable decl](https://go.dev/ref/spec#Short_variable_declarations) are being overwritten and which get masked?
I _think_ this works as expected (rewrittenSQL and err getting set without being masked, while newArgs gets created), but I've found this construction (partial redecl of anything other than err) to be a bit prone to being a future source of bugs.

I might recommend instead:
```suggestion
            var newArgs []any
			rewrittenSQL, newArgs, err = qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```

or perhaps:
```suggestion
			rewrittenSQL, args, err = qrw.RewriteQuery(context.Background(), new(pgx.Conn), sql, args)
```